lib/user.rb

```diff
@@ -0,0 +1,19 @@
+require './lib/database_connection'
+
+class User
+  def self.create(name:, username:, email:, password:)
+    result = DatabaseConnection.query(
+      "INSERT INTO users (name, username, email, password) VALUES($1, $2, $3, $4) RETURNING id, name;", 
+      [name, username, email, password]
+    )
+    User.new(id: result[0]['id'], name: result[0]['name'])
+  end
+  
+  attr_reader :id, :name
```

Comment:
I would put the attr_reader and initialise at the top of the class for visibility x